api/app/db/crud/hfi_calc.py

```diff
@@ -145,6 +145,28 @@ def store_hfi_station(session: Session, station_code: int, fuel_type_id: int, pl
     session.commit()
 
 
+def toggle_ready(session: Session,
+                 planning_area_id: int,
+                 hfi_request_id: int,
+                 username: str) -> HFIReady:
+    """ Toggles the planning area ready state for an hfi request """
+    now = get_pst_now()
+    ready_state: HFIReady = session.query(HFIReady)\
+        .filter(HFIReady.planning_area_id == planning_area_id)\
+        .filter(HFIReady.hfi_request_id == hfi_request_id)\
+        .first()
+    if ready_state.ready is True:
+        ready_state.ready = False
+    else:
+        ready_state.ready = True
+    ready_state.update_timestamp = now
+    ready_state.update_user = username
```

Comment:
Was just playing around with this in dev. Having the tooltip msg with username & timestamp of who marked it ready is a real nice detail!